catapi/domains/cat_domain.py

```diff
@@ -28,3 +28,10 @@ async def find_many(
         page=page,
     )
     return results
+
+
+async def delete_cat(cat_id: dto.CatID) -> bool:
+    deleted = await cat_model.delete_cat(cat_id=cat_id)
+    if not deleted:
+        raise exceptions.CatNotFoundError(f"Cat {cat_id} does not exist.")
+    return deleted
```

Comment:
A very nice PR! ✨ 
one small [nit] => why does function return a boolean or anything at all? 😉

Reply:
@sanjeevpc1984 I do not think we need to return anything,. Either all goes well or an error is raised.
WDYT?